src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -172,18 +175,18 @@ private void removeDeadDataNodes() {
   public void addHdfsNode(Protos.TaskID taskId, String hostname, String taskName) {
     switch (taskName) {
       case HDFSConstants.NAME_NODE_ID :
-        HashMap<String, String> nameNodes = getNameNodes();
+        Map<String, String> nameNodes = getNameNodes();
         nameNodes.put(hostname, taskId.getValue());
-        System.out.println("Saving the name node " + hostname + " " + taskId.getValue());
+        log.info("Saving the name node " + hostname + " " + taskId.getValue());
```

Comment:
Not yours, but let's log all node types or none. Maybe make them debug-level logging.